apps/prairielearn/assets/scripts/questionsTable.js

```diff
@@ -0,0 +1,190 @@
+/* eslint-env browser, jquery */
+/* global _ */
+
+const { onDocumentReady, decodeData } = require('@prairielearn/browser-utils');
+
+onDocumentReady(() => {
+  $(function () {
+    let tableSettings = {
+      icons: {
+        columns: 'fa-th-list',
+      },
+      buttons: {
+        clearFilters: {
+          text: 'Clear filters',
+          icon: 'fa-times',
+          attributes: { title: 'Clear all set question filters' },
+          event: () => {
+            $('#questionsTable').bootstrapTable('clearFilterControl');
+          },
+        },
+      },
+      onPreBody: function () {},
+      onResetView: function () {
+        $('[data-toggle="popover"]')
+          .popover({
+            sanitize: false,
+            container: 'body',
+            html: true,
+            trigger: 'hover',
+          })
+          .on('show.bs.popover', function () {
+            $($(this).data('bs.popover').getTipElement()).css('max-width', '80%');
+          });
+      },
+    };
+
+    if (decodeData('show-add-question-button')) {
```

Comment:
I count 4 calls to `decodeData` in this file. The pattern we've been using is to encode all data in a single object, decode it once, and reference all data from there.

Reply:
done